examples/jsm/loaders/GLTFLoader.js

```diff
@@ -3185,7 +3186,7 @@ class GLTFParser {
 
 				}
 
-				mesh.name = parser.createUniqueName( meshDef.name || ( 'mesh_' + meshIndex ) );
+				mesh.name = parser.createUniqueName( meshDef.name || (material && material.name) || ( 'mesh_' + meshIndex ) );
```

Comment:
What is your motivation for using the material's name for a mesh?

Reply:
I encountered the fact that I could not correctly identify materials and meshes, so I decided that a similar change would be useful for others.
I don't insist on changing it, just suggesting how I implemented it in the local script.

Reply:
Better to undo this line for now.

Can you also share a glTF asset so `texture.name = source.name;` can be verified. I was not aware that images can have names.

Reply:
I think I know what you mean. And maybe I'm misunderstanding the point of this name, but in my case it corresponds to the name of the overlay file on the geometry of the human body. 
If only a file without pictures is enough, then here: https://gist.githubusercontent.com/tony-0tis/aead36b0d213d6034e40e187431b519d/raw/fd55388a102b30e8c02e3ea38d780cf2a167e76f/gistfile1.txt

Reply:
@donmccurdy The shared glTF assets defines images like so:
```js
{
    "bufferView" : 21,
    "mimeType" : "image/jpeg",
    "name" : "RedTorsoB"
}
```
Is a `name` attribute in this context valid?

Reply:
Yes, take the torso mash for example.
Before the fix, the name of this mesh was `name: "geometry..."`, material name `name: "Torso-1"`, map mame `name: "Torso-1"`
With my fix, the mash will have a name: `"Torso-1"`, the material name: `"Torso-1"`, the mapping `name: "RedTorso"`

Reply:
Sorry, but I don't think assigning material names to meshes is correct. Better to fix this during the export.

Reply:
You are right. Apparently the gltf export plugin from blender is not fully thought out.

Reply:
> Is a `name` attribute in this context valid?

This is OK, yes — I think we do need to use `textureDef.name`, `source.name`, or both consistently regardless of the image format though, see my other comment.